src/structures/MessageAttachment.js

```diff
@@ -43,6 +43,22 @@ class MessageAttachment {
     return this;
   }
 
+  /**
+   * Sets whether this attachment is a spoiler
+   * @param {boolean} [spoiler=true] Whether the attachment should be marked as a spoiler
+   * @returns {MessageAttachment} This attachment
+   */
+  setSpoiler(spoiler = true) {
+    if (spoiler !== this.spoiler) {
+      if (spoiler) {
+        this.name = `SPOILER_${this.name}`;
+      } else {
+        this.name = this.name.replace(/^SPOILER_/, '');
+      }
+    }
+    return this;
```

Comment:
Just to remove indentation and using ternaries for simplicity:
```suggestion
    if (spoiler === this.spoiler) return this;

    this.name = spoiler ? `SPOILER_${this.name}` : this.name.replace(/^SPOILER_/, '');
    return this;
```

Also, I still prefer this variation to avoid using RegExp due to its performance implications:
```suggestion
    if (spoiler === this.spoiler) return this;

    this.name = spoiler ? `SPOILER_${this.name}` : this.name.slice('SPOILER_'.length);
    return this;
```

In runtime, doing `'SPOILER_'.length` over and over doesn't matter because since it's a constant string, it'll be optimised to just the length, so it's equally as fast as just `8` while being a lot more readable.

Reply:
Doing .slice would also mean sequential calls to setSpoiler(false) will make the name go \*poof\* tho. Now, if that's something we want to prevent is a different story

Reply:
No, because `this.spoiler` returns `true` when it's prefixed with `SPOILER_`.

If you set `spoiler` to false, and `this.spoiler` is also false (not prefixed), no operation is done.

Reply:
Brain.OS is still loading... 😅

Reply:
But the goal of removing the spoiler is only at the start, isn't it? We wouldn't want messing up the user's image if it was intentionally placed with some spoiler string, it would make more sense if it only occurred once in the *start* of the string, right? So I agree with kyranet's suggestion, except I'm a bit hesitated with regards the number 8 since Rodry's comment makes sense, regarding randomness and knowing where it comes from.

Reply:
I'm suggesting the usage of `'SPOILER_'.length` over magic numbers (`8`), not the other way around.

Reply:
Ah, got it. All good then

Reply:
You can also try a while loop that checks if the string still starts with SPOILER_ since I know some soul out there will run into this problem at some point. That way you don’t delete other words further into the name and only get rid of the spoiler itself @J-Human